core/src/main/scala/org/apache/spark/storage/BlockManagerMasterActor.scala

```diff
@@ -466,10 +467,11 @@ private[spark] class BlockManagerInfo(
       storageLevel: StorageLevel,
       memSize: Long,
       diskSize: Long,
-      tachyonSize: Long) {
+      tachyonSize: Long): BlockStatus =  {
 
-    updateLastSeenMs()
+    var blockStatus: BlockStatus = null
```

Comment:
we can get rid of this var -- just have each branch of the if/else finish w/ the block status you want to return. 

```
if (_blocks.containsKey(blockId)) {
  ...
  if (storageLevel.isValid) {
    ...
   _blocks.get(blockId)
  } else if (_blocks.containsKey(blockId)) {//I realize you aren't modifying this part, but isn't this condition redundant?
    val blockStatus: BlockStatus = _blocks.get(blockId)
    ...
    BlockStatus(storageLevel, 0, 0, 0)
  }
} else {
  null
}
```